File: app/src/main/java/com/gnoemes/shikimoriapp/entity/anime/data/AnimeResponse.java
package com.gnoemes.shikimoriapp.entity.anime.data;

import android.support.annotation.Nullable;

import com.gnoemes.shikimoriapp.entity.app.data.LinkedContentResponse;
import com.gnoemes.shikimoriapp.entity.common.data.ImageResponse;
import com.google.gson.annotations.SerializedName;

import org.joda.time.DateTime;

public class AnimeResponse extends LinkedContentResponse {

    @SerializedName("id")
    private long id;
    @SerializedName("name")
    private String name;
    @Nullable
    @SerializedName("russian")
    private String russianName;
    @SerializedName("image")
    private ImageResponse image;
    @SerializedName("url")
    private String url;
    @SerializedName("kind")
    private String type;
    @SerializedName("status")
    private String status;
    @SerializedName("episodes")
    private int episodes;
    @SerializedName("episodes_aired")
    private int episodesAired;
    @SerializedName("aired_on")
    private DateTime airedDate;
    @Nullable
    @SerializedName("released_on")
    private DateTime releasedDate;

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    @Nullable
    public String getRussianName() {
        return russianName;
    }

    public ImageResponse getImage() {
        return image;
    }

    public String getUrl() {
        return url;
    }

    public String getType() {
        return type;
    }

    public String getStatus() {
        return status;
    }

    public int getEpisodes() {
        return episodes;
    }

    public int getEpisodesAired() {
        return episodesAired;
    }

    public DateTime getAiredDate() {
        return airedDate;
    }

    @Nullable
    public DateTime getReleasedDate() {
        return releasedDate;
    }
}

File: app/src/main/java/com/gnoemes/shikimoriapp/entity/anime/data/AnimeResponse.kt
package com.gnoemes.shikimoriapp.entity.anime.data

import com.gnoemes.shikimoriapp.entity.anime.domain.AnimeStatus
import com.gnoemes.shikimoriapp.entity.anime.domain.AnimeType
import com.gnoemes.shikimoriapp.entity.app.data.LinkedContentResponse
import com.gnoemes.shikimoriapp.entity.common.data.ImageResponse
import com.google.gson.annotations.SerializedName
import org.joda.time.DateTime

data class AnimeResponse(
        @field:SerializedName("id") val id: Long,
        @field:SerializedName("name") val name: String,
        @field:SerializedName("russian") val nameRu: String?,
        @field:SerializedName("image") val image: ImageResponse,
        @field:SerializedName("url") val url: String,
        @field:SerializedName("kind") val type: AnimeType,
        @field:SerializedName("status") private val _status: AnimeStatus?,
        @field:SerializedName("episodes") val episodes: Int,
        @field:SerializedName("episodes_aired") val episodesAired: Int,
        @field:SerializedName("aired_on") val dateAired: DateTime?,
        @field:SerializedName("released_on") val dateReleased: DateTime?
) : LinkedContentResponse() {

    val status: AnimeStatus
        get() = _status ?: AnimeStatus.NONE
}